packages/wdio-browserstack-service/src/Percy/PercySDK.ts

```diff
@@ -30,7 +42,28 @@ let screenshotHandler = async (...args: any[]) => {
     PercyLogger.error('Unsupported driver for percy')
 }
 if (percySnapshot && percySnapshot.percyScreenshot) {
-    screenshotHandler = percySnapshot.percyScreenshot
+    screenshotHandler = (browser: WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser | string, screenshotName: any, options?: any) => {
+        let { name, uuid } = InsightsHandler._currentTest
+        if (!name || name === '') {
```

Comment:
let's use `isUndefined` method and create a empty check for string

Reply:
Done